ApiTestCase.php

```diff
@@ -0,0 +1,260 @@
+<?php
+
+namespace AppBundle\Test;
+
+use Coduo\PHPMatcher\Factory\SimpleFactory;
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase as BaseWebTestCase;
+use Symfony\Component\Finder\Finder;
+use Doctrine\Common\DataFixtures\Purger\ORMPurger;
+use Nelmio\Alice\Loader\Yaml as Loader;
+use Symfony\Component\HttpFoundation\Response;
+use Doctrine\ORM\EntityManagerInterface;
+
+abstract class ApiTestCase extends BaseWebTestCase
+{
+    /**
+     * @var \Symfony\Bundle\FrameworkBundle\Client $client
+     */
+    protected $client;
+
+    /**
+     * @var EntityManagerInterface
+     */
+    protected $defaultEntityManager;
+
+    public function __construct()
+    {
+        $this->defaultEntityManager = $this->get('doctrine.orm.default_entity_manager');
+    }
+
+    public function setUp()
+    {
+        $purger = new ORMPurger($this->defaultEntityManager);
+        $purger->purge();
```

Comment:
actually database immutability between tests can be achieved by starting transaction on it and rolling-back afterwards.